ckanext/dalrrd_emc_dcpr/templates/package/snippets/package_basic_fields.html

```diff
@@ -0,0 +1,129 @@
+      </div>
+    </div>
+  </div>
+  {% endblock %}
+
+  {% block package_basic_fields_org %}
+  {# if we have a default group then this wants remembering #}
+  {% if data.group_id %}
+  <input type="hidden" name="groups__0__id" value="{{ data.group_id }}" />
+  {% endif %}
+
+  {% set dataset_is_draft = data.get('state', 'draft').startswith('draft') or data.get('state', 'none') ==  'none' %}
+  {% set dataset_has_organization = data.owner_org or data.group_id %}
+  {% set organizations_available = h.organizations_available('create_dataset') %}
+  {% set user_can_update_owner_org = h.can_update_owner_org(data, organizations_available) %}
+  {% set user_is_sysadmin = h.check_access('sysadmin') %}
+  {% set show_organizations_selector = organizations_available and user_can_update_owner_org %}
+  {% set user_can_publish_dataset = h.check_access('package_publish', data) %}
+  {% set show_visibility_selector =  ( dataset_has_organization or (organizations_available and
+    (user_is_sysadmin or dataset_is_draft))) and user_can_publish_dataset %}
+
+  {% if show_organizations_selector and show_visibility_selector %}
+  <div data-module="dataset-visibility">
+    {% endif %}
+
+    {% if show_organizations_selector %}
+    {% set existing_org = data.owner_org or data.group_id %}
+    <div class="form-group control-medium">
+      <label for="field-organizations" class="control-label">{{ _('Organization') }}</label>
+      <div class="controls">
+        <select id="field-organizations" name="owner_org" data-module="autocomplete">
+          {% if h.check_config_permission('create_unowned_dataset') %}
+          <option value="" {% if not selected_org and data.id %} selected="selected" {% endif %}>{{ _('No organization') }}</option>
+          {% endif %}
+          {% for organization in organizations_available %}
+          {# get out first org from users list only if there is not an existing org #}
+          {% set selected_org = (existing_org and existing_org == organization.id) or (not existing_org and not data.id and organization.id == organizations_available[0].id) %}
+          <option value="{{ organization.id }}" {% if selected_org %} selected="selected" {% endif %}>{{ organization.display_name }}</option>
+          {% endfor %}
+        </select>
+      </div>
+    </div>
+    {% endif %}
+
+    {% if show_visibility_selector %}
+    {% block package_metadata_fields_visibility %}
+    <div class="form-group control-medium">
+      <label for="field-private" class="control-label">{{ _('Visibility') }}</label>
+      <div class="controls">
+        <select id="field-private" name="private" class="form-control">
+          {% for option in [('True', _('Private')), ('False', _('Public'))] %}
+          <option value="{{ option[0] }}" {% if option[0] == data.private|trim %}selected="selected"{% endif %}>{{ option[1] }}</option>
+          {% endfor %}
+        </select>
+      </div>
+    </div>
+    {% endblock %}
+    {% endif %}
```

Comment:
As mentioned in the comment above, lets add a branch that adds the `private` field as a hidden input type when `show_visibility_selector` is False